assets/js/modules/analytics/settings/settings-main.js

```diff
@@ -45,9 +47,11 @@ export default function SettingsMain( { isOpen, isEditing } ) {
 		return null;
 	}
 
-	if ( isEditing ) {
-		return <SettingsEdit />;
-	}
-
-	return <SettingsView />;
+	return (
+		<RestoreSnapshots>
+			<PermissionsModal dataStoreToSnapshot={ STORE_NAME } />
+			{ isEditing && <SettingsEdit /> }
+			{ ! isEditing && <SettingsView /> }
```

Comment:
It might be that we need to scope the `<RestoreSnapshots>` code into this component, because otherwise we get settings rollback. I'll try that quickly and report back.

Reply:
Argh, sorry, I meant wrapping `SettingsEdit`, not `SettingsView`.